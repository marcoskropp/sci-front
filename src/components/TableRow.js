import React from 'react';
import PropTypes from 'prop-types';
import {
  Checkbox,
  TableRow as TableRowMui,
  TableCell
} from '@material-ui/core';

function TableRow(props) {
  const {
    title,
    description,
    place,
    startDate,
    endDate,
    subscribed,
    checkWorkshop
  } = props;

  return (
    <TableRowMui>
      <TableCell>
        <Checkbox name={title} onClick={checkWorkshop} checked={subscribed} />
      </TableCell>
      <TableCell>{title}</TableCell>
      <TableCell>{description}</TableCell>
      <TableCell>{place}</TableCell>
      <TableCell>{new Date(startDate).toLocaleDateString()}</TableCell>
      <TableCell>{new Date(endDate).toLocaleDateString()}</TableCell>
    </TableRowMui>
  );
}

TableRow.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  place: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  subscribed: PropTypes.bool,
  checkWorkshop: PropTypes.func
};

export default TableRow;
