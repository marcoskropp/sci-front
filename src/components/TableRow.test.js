import React from 'react';
import { shallow } from 'enzyme';
import { name, lorem, random } from 'faker';

import TableRow from './TableRow';

const props = {
  title: name.title(),
  description: lorem.words(),
  place: lorem.word(),
  startDate: lorem.word(),
  endDate: lorem.word(),
  subscribed: random.boolean(),
  checkWorkshop: () => {}
};

describe('Testing TableRow Component', () => {
  it('Render Component', () => {
    const wrapper = shallow(<TableRow {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
