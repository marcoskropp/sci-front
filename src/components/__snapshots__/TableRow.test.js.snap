// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TableRow Component Render Component 1`] = `ShallowWrapper {}`;
